xapian-core/include/xapian/mset.h

```diff
@@ -74,6 +77,30 @@ class XAPIAN_VISIBILITY_DEFAULT MSet {
     /// Destructor.
     ~MSet();
 
+    /** Assigns new weights to items
+     * This functon is used in letor to update the weights assigned by the ranker
+     * It updates weights of all the elements in items as well as updating the max_attained
```

Comment:
` /** Assigns new weights and updates MSet.
     *
     *  Dereferencing the Iterator should return a double.
     *
     *  Iterator should dereference weights in the original order of documents.
     *  Weights are assigned as per the original order of documents.
     *
     *  @param begin	Begin iterator.
     *  @param end	End iterator.
     *
     *  @exception Xapian::InvalidArgument is thrown if the total number of
     *		    elements in the input doesn't match the total number of
     *		    documents in MSet.
     */`
will be using this docstring if it is acceptable to you